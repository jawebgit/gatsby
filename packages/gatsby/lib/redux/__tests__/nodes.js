const { actions, boundActions } = require("../actions")
const nodeReducer = require("../reducers/nodes")

describe(`Create and update nodes`, () => {
  // TODO add these back when we stop directly consoleing errors.
  // Right now makes tests noisy.
  //
  // it(`validates created nodes`, () => {
  // const action = actions.createNode({
  // type: `Test`,
  // });
  // expect(action.type).toEqual(`VALIDATION_ERROR`);
  // });
  // it(`validates updated nodes`, () => {
  // const action = actions.updateNode({
  // type: `Test`,
  // });
  // expect(action.type).toEqual(`VALIDATION_ERROR`);
  // });

  it(`allows creating nodes`, () => {
    const action = actions.createNode({
      id: `hi`,
      contentDigest: `hasdfljds`,
      children: [],
      parent: `test`,
      mediaType: `test`,
      pluginName: `tests`,
      type: `Test`,
      pickle: true,
    })
    expect(action).toMatchSnapshot()
    expect(nodeReducer(undefined, action)).toMatchSnapshot()
  })

  it(`allows updating nodes`, () => {
    const action = actions.createNode({
      id: `hi`,
      contentDigest: `hasdfljds`,
      children: [],
      parent: `test`,
      mediaType: `test`,
      pluginName: `tests`,
      type: `Test`,
      pickle: true,
    })
    const updateAction = actions.createNode({
      id: `hi`,
      contentDigest: `hasdfljds`,
      children: [],
      parent: `test`,
      mediaType: `test`,
      pluginName: `tests`,
      type: `Test`,
      pickle: false,
    })
    let state = nodeReducer(undefined, action)
    state = nodeReducer(state, updateAction)
    expect(state["hi"].pickle).toEqual(false)
  })
})
