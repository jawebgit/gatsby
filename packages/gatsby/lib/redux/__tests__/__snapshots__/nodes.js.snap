// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create and update nodes allows creating nodes 1`] = `
Object {
  "payload": Object {
    "children": Array [],
    "contentDigest": "hasdfljds",
    "id": "hi",
    "mediaType": "test",
    "parent": "test",
    "pickle": true,
    "pluginName": "tests",
    "type": "Test",
  },
  "type": "CREATE_NODE",
}
`;

exports[`Create and update nodes allows creating nodes 2`] = `
Object {
  "hi": Object {
    "children": Array [],
    "contentDigest": "hasdfljds",
    "id": "hi",
    "mediaType": "test",
    "parent": "test",
    "pickle": true,
    "pluginName": "tests",
    "type": "Test",
  },
}
`;
